tests/test_crash_fault_tolerance.sh

```diff
@@ -20,29 +20,51 @@
 # a node, verifying all nodes reach block 20, restarting the node, and
 # verifying that all nodes reach block 30.
 
-if [ -z "$ISOLATION_ID " ]; then export ISOLATION_ID=latest; fi
+if [ -z "$ISOLATION_ID" ]; then export ISOLATION_ID=latest; fi
```

Comment:
I don't think this commit is necessary, since we can already run the tests locally with `bin/run_docker_test` which is the recommended way of running it. Also, it makes the output of the test rather messy, and from my experience the containers aren't cleaned up properly when using `run_docker_test`

Reply:
I really think we should move away from the docker-in-docker thing, I don't see the value-add for the complexity that it brings.

Reply:
@aludvik could you weigh in on this?

Reply:
@ltseeley I think this commit has a number of changes that are good, regardless of whether the test is ultimately run under docker or with run_docker_test. I would not say they are unnecessary and I don't think using `bin/run_docker_test` fixes the problems.

@knkski I would like to hear more about your opinions on why docker is bad and what you think some better alternatives are. I agree it can get complex, but the state of things is derived from a number of somewhat conflicting requirements on how we build up our integration tests such as:
1. Must be able to run the tests from any host, with a minimal set of requirements (usually docker, bash, and python).
2. Must be able to run the tests on Jenkins without separate jobs and tests interfering with each other.
3. Must not do docker-in-docker (this is why we mount the docker socket into the container, it is more of a docker-alongside-docker solution, I think this is the article I read when coming up with the solution: https://jpetazzo.github.io/2015/09/03/do-not-use-docker-in-docker-for-ci/)